src/server/journal/streamer.cc

```diff
@@ -317,37 +318,46 @@ void RestoreStreamer::OnDbChange(DbIndex db_index, const DbSlice::ChangeReq& req
 
 void RestoreStreamer::WriteEntry(string_view key, const PrimeValue& pk, const PrimeValue& pv,
                                  uint64_t expire_ms) {
-  absl::InlinedVector<string_view, 5> args;
-  args.push_back(key);
-
-  string expire_str = absl::StrCat(expire_ms);
-  args.push_back(expire_str);
-
-  io::StringSink restore_cmd_sink;
-  {  // to destroy extra copy
-    io::StringSink value_dump_sink;
-    SerializerBase::DumpObject(pv, &value_dump_sink);
-    args.push_back(value_dump_sink.str());
-
-    args.push_back("ABSTTL");  // Means expire string is since epoch
-
-    if (pk.IsSticky()) {
-      args.push_back("STICK");
+  // We send RESTORE commands for small objects, or objects we don't support breaking.
+  CmdSerializer serializer([&](std::string s) { Write(s); });
+
+  bool use_restore_serialization = true;
```

Comment:
all this logic below should be in your new class CmdSerializer

Reply:
The only api to use it will be SerializeEntry which will get the key value and expire time